### PR TITLE
[flink]  Support add hive catalog table properties

### LINKED ISSUE #1983: [Feature] CDC Ingestion support configuring hive table properties
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

CDC Ingestion support configuring hive table properties
for example：table owner, life cycle...

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

close #1983

### Tests

org.apache.paimon.flink.action.cdc.ActionFactoryTest

### Documentation

cdc-ingestion#Adding Attributes to a Hive Table

### CODE REVIEW

**Comment by JingsongLi** [comment]
First, I think this should be a feature for `HiveCatalog` instead of CDC ingestion.

**Comment by JingsongLi** [comment]
Can we just put all table options into Hive `parameters`?

**Comment by MonsterChenzhuo** [comment]
> Can we just put all table options into Hive `parameters`?

**Comment by MonsterChenzhuo** [comment]
> Can we just put all table options into Hive `parameters`?

Yes, but what we hope is to automatically add it during the synchronization process.

**Comment by JingsongLi** [comment]
Before creating tables, you can add these table conf?

**Comment by MonsterChenzhuo** [comment]
> 

I cannot add it because the CDC Ingestion process already includes the automatic creation of Hive tables.

**Comment by MonsterChenzhuo** [comment]
> Before creating tables, you can add these table conf?

I cannot add it because the CDC Ingestion process already includes the automatic creation of Hive tables.

File: docs/content/how-to/cdc-ingestion.md [hunk 1]
```diff
@@ -692,6 +692,16 @@ you can specify type mapping option `tinyint1-not-bool` (Use `--type-mapping`),
 6. MySQL BINARY will be mapped to Paimon VARBINARY. This is because the binary value is passed as bytes in binlog, so it 
 should be mapped to byte type (BYTES or VARBINARY). We choose VARBINARY because it can retain the length information.
 
+## Adding Properties to a Hive Table
```
Comment (JingsongLi) [inline docs/content/how-to/cdc-ingestion.md:695]: Don't add here, document this in create-catalogs.md, and for example, after `Synchronizing Partitions into Hive Metastore`.

File: docs/content/how-to/cdc-ingestion.md [hunk 2]
```diff
@@ -692,6 +692,16 @@ you can specify type mapping option `tinyint1-not-bool` (Use `--type-mapping`),
...
+
+Using the --table-conf parameter allows for the convenient definition of Hive Table properties. When you utilize configurations
+that start with “hive.”, these configurations will be defined in the “TBLPROPERTIES” of the Hive Table, thereby specifying the table's attributes.
+for example:
+```bash
+--table-conf hive.table.owner=Jon
+```
+In the command mentioned above, hive.table.owner=Jon will be used to add to the Hive table attributes during automatic table creation.
```
Comment (JingsongLi) [inline docs/content/how-to/cdc-ingestion.md:703]: `table.owner=Jon` will be added to ....?

File: docs/content/how-to/cdc-ingestion.md (hunk 1 above)
Comment (MonsterChenzhuo) [reply docs/content/how-to/cdc-ingestion.md]: done.

**Review by JingsongLi — COMMENTED** [review]
You might be misunderstanding something, I mean this pr should have nothing to do with cdc at all, no code changes from cdc.

This feature should just be implemented by HiveCatalog.

File: docs/content/how-to/creating-catalogs.md [hunk 3]
```diff
@@ -163,3 +163,16 @@ You can set location in the properties of table/database by the config of `locat
 By default, Paimon does not synchronize newly created partitions into Hive metastore. Users will see an unpartitioned table in Hive. Partition push-down will be carried out by filter push-down instead.
 
 If you want to see a partitioned table in Hive and also synchronize newly created partitions into Hive metastore, please set the table property `metastore.partitioned-table` to true. Also see [CoreOptions]({{< ref "maintenance/configurations#CoreOptions" >}}).
+
+### Adding Parameters to a Hive Table
+
+Using the `--table-conf` parameter allows for the convenient definition of Hive table parameters. When you utilize configurations
+that start with “hive.”, these configurations will be defined in the `TBLPROPERTIES` of the Hive Table.
+for example:
+```bash
```
Comment (JingsongLi) [inline docs/content/how-to/creating-catalogs.md:172]: remove this example, you can just say "if you add table option `hive.table.owner`, then ....."

File: docs/content/how-to/creating-catalogs.md [hunk 4]
```diff
@@ -163,3 +163,16 @@ You can set location in the properties of table/database by the config of `locat
 By default, Paimon does not synchronize newly created partitions into Hive metastore. Users will see an unpartitioned table in Hive. Partition push-down will be carried out by filter push-down instead.
 
 If you want to see a partitioned table in Hive and also synchronize newly created partitions into Hive metastore, please set the table property `metastore.partitioned-table` to true. Also see [CoreOptions]({{< ref "maintenance/configurations#CoreOptions" >}}).
+
+### Adding Parameters to a Hive Table
+
+Using the `--table-conf` parameter allows for the convenient definition of Hive table parameters. When you utilize configurations
```
Comment (JingsongLi) [inline docs/content/how-to/creating-catalogs.md:169]: table option, not mention `--table-conf`

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 5]
```diff
@@ -324,7 +324,15 @@ public void createTable(Identifier identifier, Schema schema, boolean ignoreIfEx
                             + " to underlying files.",
                     e);
         }
-        Table table = newHmsTable(identifier);
+        Table table =
+                newHmsTable(
+                        identifier,
+                        tableSchema.options().entrySet().stream()
+                                .filter(entry -> entry.getKey().startsWith("hive."))
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:331]: Use static final string.
substring(5) also

File: docs/content/how-to/creating-catalogs.md (hunk 4 above)
Comment (MonsterChenzhuo) [reply docs/content/how-to/creating-catalogs.md]: done.

File: docs/content/how-to/creating-catalogs.md (hunk 3 above)
Comment (MonsterChenzhuo) [reply docs/content/how-to/creating-catalogs.md]: done.

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java (hunk 5 above)
Comment (MonsterChenzhuo) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java]: done.

File: paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java [hunk 6]
```diff
@@ -158,6 +159,19 @@ default boolean tableExists(Identifier identifier) {
...
+     * Retrieve the properties of a Hive table.
+     *
+     * <p>NOTE: Properties of system tables cannot be retrieved.
+     *
+     * @param identifier The identifier of the table whose properties are to be retrieved.
+     * @return A map containing the properties of the table.
+     * @throws TableNotExistException If no table exists with the given identifier.
+     * @throws DatabaseNotExistException If the database specified in the identifier does not exist.
+     */
+    Map<String, String> getTableParameters(Identifier identifier)
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:172]: This PUBLIC interface is only for testing. And it is only for Hive.

You should remove it and get Hive table from Hive api in testing.

Comment (MonsterChenzhuo) [reply paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java]: done.

**Review by JingsongLi — APPROVED** [review]
+1

**Comment by JingsongLi** [comment]
Thanks @MonsterChenzhuo 
Merging...

**Pull request merged**